diff --git a/includes/bootstrap.inc b/includes/bootstrap.inc
index bc5d5a3..c022e8e 100644
--- a/includes/bootstrap.inc
+++ b/includes/bootstrap.inc
@@ -2519,7 +2519,6 @@ function get_t() {
  * Initialize all the defined language types.
  */
 function drupal_language_initialize() {
-  global $language;
   $types = language_types();
 
   // Ensure the language is correctly returned, even without multilanguage
@@ -2539,9 +2538,6 @@ function drupal_language_initialize() {
     // environments.
     bootstrap_invoke_all('language_init');
   }
-
-  // Send appropriate HTTP-Header for browsers and search engines.
-  header('Content-Language: ' . $language->language);
 }
 
 /**
diff --git a/includes/common.inc b/includes/common.inc
index d34c19d..c91f87d 100644
--- a/includes/common.inc
+++ b/includes/common.inc
@@ -2467,6 +2467,10 @@ function drupal_deliver_html_page($page_callback_result) {
     drupal_add_http_header('Content-Type', 'text/html; charset=utf-8');
   }
 
+  // Send appropriate HTTP-Header for browsers and search engines.
+  global $language;
+  drupal_add_http_header('Content-Language', $language->language);
+
   // Menu status constants are integers; page content is a string or array.
   if (is_int($page_callback_result)) {
     // @todo: Break these up into separate functions?
